flytekit/types/directory/types.py

```diff
@@ -236,6 +236,66 @@ def new_dir(self, name: typing.Optional[str] = None) -> FlyteDirectory:
     def download(self) -> str:
         return self.__fspath__()
 
+    @classmethod
+    def listdir(cls, directory: FlyteDirectory) -> typing.List[typing.Union[FlyteDirectory, FlyteFile]]:
```

Comment:
why not make this as an instance method on FlyteDirectory instead of class method?
Also i want to support async versions of all these methods

Reply:
I can add them to a separate PR.